File: Java/PerformanceTestSequential.java
// Total Execution Time: 3.249 seconds
// This sequential execution is almost 2 times slower than by creating two separate threads of execution in PerformanceTestThreaded.java

public class PerformanceTestSequential
{
	int N;
	
	public PerformanceTestSequential(int n)
	{
		this.N = n;
	}
	
	public void run()
	{
		while(this.N > 0)
		{
			this.N -= 1;
		}
	}
	
	public static void main(String args[])
	{
		PerformanceTestSequential t1 = new PerformanceTestSequential(2000000000);
		PerformanceTestSequential t2 = new PerformanceTestSequential(2000000000);
		
		long startTime = System.nanoTime();
		t1.run();
		t2.run();
		long endTime   = System.nanoTime();
		long totalTime = endTime - startTime;
		
		System.out.println("\n\nExecution Time: " + (double)totalTime / 1000000000);
	}
}

// System Used-
// OS: Windows 10; 64-bit Operating System
// Processor: 2.53 GHz Intel Core i3
// RAM: 2GB
// Cores: 2
// Logical Processors: 4

File: PHP/performance_test_sequential.php
<?php

// Total Execution Time of 'performance_test_threaded.php' = 14.210937023163 Secs
// Total Execution Time of 'performance_test_sequential.php' = 7.1525769233704 Secs

class Sequential {
	
	public $N;
	public function __construct($number)
	{
		$this->N = $number;
	}
	
	public function run()
	{
		while($this->N > 0)
		{
			$this->N = $this->N - 1;
		}
	}
}

$start_time = microtime(true);
$object1 = new Sequential(10000000);
$object2 = new Sequential(10000000);
$object1->run();
$object2->run();
$end_time = microtime(true);

$total_time = ($end_time - $start_time);
echo '<b>Total Execution Time:</b> '.$total_time.' Secs';
?>
